src/libstd/thread/block_on_future.rs

```diff
@@ -0,0 +1,234 @@
+//! Future/Task/Thread integration.
+//! The method defined in this module allows to block a thread until an
+//! async task completes.
+
+use crate::future::Future;
+use crate::task::{Context, Poll, RawWaker, RawWakerVTable, Waker};
+use crate::mem;
+use crate::pin::Pin;
+use crate::sync::Arc;
+use super::{current, park, Inner, Thread};
+
+/// Carries a flag that is used to wakeup the executor.
+/// A pointer to this struct is passed to the thread-local waker.
+struct LocalWakeState {
+    is_woken: bool,
+    waker_was_cloned: bool,
+}
+
+/// Returns the vtable that is used for waking up the executor
+/// from another thread.
+fn threadsafe_waker_vtable() -> &'static RawWakerVTable {
+    &RawWakerVTable::new(
+        clone_threadsafe_waker,
+        wake_threadsafe_waker,
+        wake_threadsafe_waker_by_ref,
+        drop_threadsafe_waker,
+    )
+}
+
+/// Returns the vtable that is used for waking up the executor
+/// from inside it's execution on the current thread.
+fn current_thread_waker_vtable() -> &'static RawWakerVTable {
+    &RawWakerVTable::new(
+        create_threadsafe_waker,
+        wake_current_thread,
+        wake_current_thread_by_ref,
+        |_| {},
+    )
+}
+
+/// This method will be called when the waker reference gets cloned,
+/// which makes it possible to transfer it to another thread. In this
+/// case we have to create a threadsafe `Waker`. In order to to this
+/// we retain the thread handle and store it in the new `RawWaker`s
+/// data pointer.
+unsafe fn create_threadsafe_waker(data: *const()) -> RawWaker {
+    let wake_state = data as *mut LocalWakeState;
+    (*wake_state).waker_was_cloned = true;
+
+    // Get the Arc<Inner> of a current thread handle and store into in
```

Comment:
General stylistic note: the module sometimes uses backticks and sometimes not when talking about things from code -- can you go through the module and use backticks everywhere consistently?

Reply:
True. I'm not really sure if backticks in comments bring that much value, since those things are never pretty-printed and the backticks are annoying to type (at least on a german keyboard layout). Therefore I guess I added them only in some places.

I now added them to the remaining places where types are referenced.